tests/unit/objectives/test_multi_objectives.py

```diff
@@ -137,33 +185,40 @@ def test_gen_pareto_front_is_equal_to_math_defined(
         (VLMOP2(2), tf.constant([[0.4, 0.2, 0.5]])),
         (DTLZ1(3, 2), tf.constant([[0.3, 0.1]])),
         (DTLZ2(5, 2), tf.constant([[0.3, 0.1]])),
+        (ConstrainedBraninCurrin(), tf.constant([[0.3, 0.2, 0.1]])),
     ],
 )
 def test_func_raises_specified_input_dim_not_align_with_actual_input_dim(
-    obj_inst: MultiObjectiveTestProblem, actual_x: TensorType
+    obj_inst: Union[MultiObjectiveTestProblem, ConstrainedMultiObjectiveTestProblem],
+    actual_x: TensorType,
 ) -> None:
     with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
         obj_inst.objective(actual_x)
+    if isinstance(obj_inst, ConstrainedMultiObjectiveTestProblem):
+        with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+            obj_inst.constraint(actual_x)
 
 
 @pytest.mark.parametrize(
-    "problem, input_dim, num_obj",
+    "problem, input_dim, num_obj, num_con",
     [
-        (VLMOP2(2), 2, 2),
-        (VLMOP2(10), 10, 2),
-        (DTLZ1(3, 2), 3, 2),
-        (DTLZ1(10, 5), 10, 5),
-        (DTLZ2(3, 2), 3, 2),
-        (DTLZ2(10, 5), 10, 5),
+        (VLMOP2(2), 2, 2, 0),
+        (VLMOP2(10), 10, 2, 0),
+        (DTLZ1(3, 2), 3, 2, 0),
+        (DTLZ1(10, 5), 10, 5, 0),
+        (DTLZ2(3, 2), 3, 2, 0),
+        (DTLZ2(10, 5), 10, 5, 0),
+        (ConstrainedBraninCurrin(), 2, 2, 1),
     ],
 )
 @pytest.mark.parametrize("num_obs", [1, 5, 10])
 @pytest.mark.parametrize("dtype", [tf.float32, tf.float64])
-def test_objective_has_correct_shape_and_dtype(
-    problem: MultiObjectiveTestProblem,
+def test_objective_and_constraint_has_correct_shape_and_dtype(
+    problem: Union[MultiObjectiveTestProblem, ConstrainedMultiObjectiveTestProblem],
```

Comment:
Is this change necessary given ConstrainedMultiObjectiveTestProblem is a subclass of MultiObjectiveTestProblem?

Reply:
I agree, probably not